control-plane/agents/src/common/errors.rs

```diff
@@ -32,13 +32,21 @@ pub enum SvcError {
     #[snafu(display("Node {} is cordoned", node_id))]
     CordonedNode { node_id: String },
     #[snafu(display("Node {node_id} is already labelled with labels '{labels}'"))]
-    LabelsExists {
+    NodeLabelsExists {
         node_id: String,
         labels: String,
         conflict: bool,
     },
+    #[snafu(display("Pool {pool_id} is already labelled with labels '{labels}'"))]
+    PoolLabelsExists {
+        pool_id: String,
+        labels: String,
+        conflict: bool,
+    },
```

Comment:
hmm maybe this will help remove duplicate errors?

```suggestion
    #[snafu(display("{resource} {id} is already labelled with labels '{labels}'"))]
    LabelsExists {
        resource: ResourceKind,
        id: String,
        labels: String,
        conflict: bool,
    },
```

Reply:
This can reduce the code but when we map to to `ReplyError`  we need a resource kind. 
```
SvcError::PoolLabelsExists { conflict, .. } => ReplyError {
                kind: if conflict {
                    ReplyErrorKind::FailedPrecondition
                } else {
                    ReplyErrorKind::AlreadyExists
                },
                resource: ResourceKind::Pool,
                source,
                extra,
            },
  ```

Reply:
Which would be in resource field